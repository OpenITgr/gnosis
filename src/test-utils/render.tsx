/* eslint-disable import/export */
import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
