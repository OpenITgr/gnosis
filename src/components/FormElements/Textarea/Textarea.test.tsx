import React from "react";
import userEvent from "@testing-library/user-event";
import { Textarea } from "./Textarea.stories";
import { screen, render } from "@test-utils/render";

describe("<Textarea />", () => {
  it("renders correctly", () => {
    render(<Textarea {...Textarea.args} />);

    const textarea = screen.getByLabelText("Description");

    expect(textarea).toBeInTheDocument();
    expect(textarea).toHaveAttribute("placeholder", "Type some description");
  });

  it("changes value", () => {
    render(<Textarea {...Textarea.args} />);

    const textarea = screen.getByLabelText("Description");

    expect(textarea).toHaveValue("");

    userEvent.type(textarea, "Test value");

    expect(textarea).toHaveValue("Test value");
  });

  it("matches snapshot", () => {
    const { container } = render(<Textarea {...Textarea.args} />);

    expect(container).toMatchSnapshot();
  });
});
