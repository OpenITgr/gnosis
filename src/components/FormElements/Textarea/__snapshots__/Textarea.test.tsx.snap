// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> matches snapshot 1`] = `
<div>
  <div
    class="valid css-1f6dkts-inputContainerBaseStyles-inputContainerBaseStyles-inputBaseStyles-inputBaseStyles-textareaContainer-textareaContainer"
  >
    <label
      class="css-xs6o3g-label-label"
      for="description"
    >
      Description
    </label>
    <textarea
      id="description"
      placeholder="Type some description"
    />
  </div>
</div>
`;
